### PR TITLE
Kubernetes get started

### PR DESCRIPTION
**Pull request by zaibon**

### CODE REVIEW

File: docs/kubernetes/getstarted.md [hunk 1]
```diff
@@ -0,0 +1,604 @@
...
+In this guide we will walk you through the provisioning of a full-blown kubernetes cluster
+on the TF grid.
+
+We will then see how to connect to it and interact using kubectl on our local machine.
+
+Finally we will go through some examples use cases to grasp the features offered by the cluster.
+
+# TL;DR
+
+If you quickly want to setup a kubernetes cluster and start to play with it, you can launch the startup script.
```
Comment (zaibon) [inline docs/kubernetes/getstarted.md]: which startup script ? maybe make a link to the actual file.

File: docs/kubernetes/getstarted.md [hunk 2]
```diff
@@ -0,0 +1,604 @@
...
+
+Prerequisite:
+
+- linux based OS
+- [kubectl](https://kubernetes.io/fr/docs/tasks/tools/install-kubectl/)
+- github account with [ssh key linked to your account](https://help.github.com/en/enterprise/2.17/user/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
```
Comment (zaibon) [inline docs/kubernetes/getstarted.md]: This is not requried, you can perfectly just create VM and specify an SSH public key in the reservation. 
Github username is just a "trick" k3os provide, but we should not advertise it as the main way of authorizing using into their VM.

File: docs/kubernetes/getstarted.md [hunk 3]
```diff
@@ -0,0 +1,604 @@
...
+```
+$ chmod +x startup.sh
+$ ./startup.sh zgorizzo69 "173.40" 1 2
+```
+
+This will build tfuser provision a network
+
+# Provisioning and Deploiement of the cluster
+
+Our CLI tool tfuser makes it easy to interact with the TFGrid and provision a kubernetes cluster.
```
Comment (zaibon) [inline docs/kubernetes/getstarted.md]: tfuser is only a developer tool and only works on `devnet`. Let's be very explicit about that.

File: docs/kubernetes/getstarted.md [hunk 4]
```diff
@@ -0,0 +1,604 @@
...
+$ cd tools
+$ make tfuser
+cd tfuser && go build -ldflags '-w -s -X github.com/threefoldtech/zos/pkg/version.Branch=fcvm -X github.com/threefoldtech/zos/pkg/version.Revision=c00740b8fbb75746cd02fb5614ccf57342b7a9f4 -X github.com/threefoldtech/zos/pkg/version.Dirty='
+$ cd tfuser
+$ ./tfuser --help
+```
+
+## Identity on the TF Grid
+
+To interact with the grid we need an indentity which is simply some cryptographic keys that we will use to sign transaction on the network.
```
Comment (zaibon) [inline docs/kubernetes/getstarted.md]: Here again this is only valid for devnet.

File: docs/kubernetes/getstarted.md [hunk 5]
```diff
@@ -0,0 +1,604 @@
...
+new identity generated: 7HZd49ugHzJxQUnHKyQRbmcwWWvKKtoJUndfiiXGTkow
+seed saved at user.seed
+```
+
+NB: a file named user.seed has been created. This file is sensitive as it stores your private key. Make sure not to loose that file.
+
+## Provisioning
+
+To provision our VMs we first need to setup a network between different Nodes on the TFGrid
+We will use devnet Nodes to select some nodes go to :[devnet Cockpit](https://cockpit.devnet.grid.tf/)
```
Comment (zaibon) [inline docs/kubernetes/getstarted.md]: If they want an API to explore the nodes on devnet, they can use https://explorer.devnet.grid.tf/nodes

File: docs/kubernetes/getstarted.md [hunk 6]
```diff
@@ -0,0 +1,604 @@
...
+
+- one for the global network with a cidr /16
+- one for network resources on the node with a cidr /24. All workloads in this network resource will receive an IP from this subnet. The subnet must be part of the CIDR provided when the network is created.
+
+**NOTE** that you can't give the same subnet to all of your nodes. Each node must be assigned a different /24 subnet that is part of the global /16 of the network.
+
+We will need an external access to the network to be able to ssh into our VMs.
+
+```
+$ ./tfuser generate --schema network.json network add-access --node qzuTJJVd5boi6Uyoco1WWnSgzTb7q8uN79AjBT9x9N3 \
```
Comment (zaibon) [inline docs/kubernetes/getstarted.md]: Maybe a note regarding how to pick an entry point node here.
If user doesn't have IPv6, then they have to make sure to pick a node that has a public IPv4 address and use the `--ip4` flag.
For user with IPv6, the `--ip4` flag is not required

File: docs/kubernetes/getstarted.md [hunk 7]
```diff
@@ -0,0 +1,604 @@
...
+
+```
+$ ./tfuser generate --schema network.json network graph
+```
+
+This will generate network.json.dot file. You can copy paste the content on a website like [graphiz](https://dreampuf.github.io/GraphvizOnline) to see the visual representation of the network we defined.
+
+![network graphical representation](network_dot_file.png)
+
+Now we have a nodes in a network definition ready to be provisioned. Let's use our seed to sign the transaction and ask for a 2 hours provision.
```
Comment (zaibon) [inline docs/kubernetes/getstarted.md]: `--duration 2` doesn't reserve for 2 hours but for 2 days. If you want 2hours then you need to do `--duration 120m`

File: docs/kubernetes/getstarted.md [hunk 8]
```diff
@@ -0,0 +1,604 @@
...
+$ wg-quick up ./wg.conf
+```
+
+We have setup and provision a network on the grid that we can join through wireguard.
+Let's now provision some Kubernetes VM on those nodes
+
+### Provision
+
+Let's provision a kubernetes VM on our first node and assign an IP allowed in the network `kubetest` defined here above.
+Replace `GITHUB_USERNAME` by your github username. We choose an IP for the VM that is part of the subnet we previously defined for the node. We can't choose 173.30.3.1 as it is the IP of the network ressource itself.
```
Comment (zaibon) [inline docs/kubernetes/getstarted.md]: Instead of using github username let's show how they can just give their public SSH key here. And as a note, example github username is possible.

File: docs/kubernetes/getstarted.md [hunk 9]
```diff
@@ -0,0 +1,604 @@
...
+```
+
+We can browse localhost on port 8888 to find the grafana UI e.g. [http://localhost:8888/](http://localhost:8888/)
+The username is `admin` and the default admin password to log into the grafana UI is `prom-operator`
+Then you can for instance import a dashboard and use the ID 8588 and don't forget to select the prometheus data source
+![kubernetes-deployment-statefulset-daemonset-metrics dashboard](ressources/grafana/grafana1.png)
+or the dashboard ID 6879
+![analysis-by-pod](ressources/grafana/grafana2.png)
+
+## Storage
```
Comment (zaibon) [inline docs/kubernetes/getstarted.md]: I think everything after this line should be moved out of the getting started. This is very specific and more deep topics that should not be part of a getting started.

File: docs/kubernetes/ressources/README.md [hunk 10]
```diff
@@ -0,0 +1,539 @@
+# K3s on TF Grid
```
Comment (zaibon) [inline docs/kubernetes/ressources/README.md]: Is this file still needed now that we have the `Getting started` ? It seems it is a bit of a duplicate

File: docs/kubernetes/getstarted.md (hunk 1 above)
Comment (zgorizzo69) [reply docs/kubernetes/getstarted.md]: I added a comment

File: docs/kubernetes/getstarted.md (hunk 2 above)
Comment (zgorizzo69) [reply docs/kubernetes/getstarted.md]: yep I modified the script accordingly

File: docs/kubernetes/getstarted.md (hunk 4 above)
Comment (zgorizzo69) [reply docs/kubernetes/getstarted.md]: I added a disclaimer

File: docs/kubernetes/getstarted.md (hunk 5 above)
Comment (zgorizzo69) [reply docs/kubernetes/getstarted.md]: comment added

File: docs/kubernetes/getstarted.md (hunk 7 above)
Comment (zgorizzo69) [reply docs/kubernetes/getstarted.md]: mistake from copy paste .. corrected

File: docs/kubernetes/getstarted.md (hunk 6 above)
Comment (zgorizzo69) [reply docs/kubernetes/getstarted.md]: probably but for now the only one working for an external access was node : qzuTJJVd5boi6Uyoco1WWnSgzTb7q8uN79AjBT9x9N3.

 I would love to try with a node that has a pulbic IPv6 address . Do you know a node that can work  @zaibon ?

Comment (zaibon) [reply docs/kubernetes/getstarted.md]: > I would love to try with a node that has a pulbic IPv6 address . Do you know a node that can work @zaibon ?

All the nodes have ipv6. So you should be able to pick any.

File: docs/kubernetes/getstarted.md (hunk 8 above)
Comment (zgorizzo69) [reply docs/kubernetes/getstarted.md]: done

File: docs/kubernetes/getstarted.md (hunk 9 above)
Comment (zgorizzo69) [reply docs/kubernetes/getstarted.md]: created an advanced doc and added a link to it

File: docs/kubernetes/ressources/README.md (hunk 10 above)
Comment (zgorizzo69) [reply docs/kubernetes/ressources/README.md]: right ..deleted
